api/controllers/api-user.js

```diff
@@ -1,8 +1,32 @@
 'use strict'
 
+/* global fetch */
 const ApiUser = require('../../models/api-user')
-const https = require('https')
-const xml2js = require('xml2js')
+
+// Lookup up a user by the email address and return a promise that
+// resolves to their The Key guid if they exist, or rejects if they don't
+async function lookupUser (emailAddress) {
```

Comment:
I wonder if this will be on the slow side since it seems to query against all of the users. I wonder if it would be better to do it the way cortex gateway does it:

```
GET https://signon.okta.com/oauth2/v1/userinfo
  Authorization: Bearer <access-token-from-client>

parse JSON response body to get ssoGuid from it
```

Reply:
This method needs to lookup an arbitrary user by their email address. The `/userinfo` endpoint can only lookup the profile of the logged in user (based on their access token).

Reply:
I see. I thought this was only used at login time. So. it is also used during user creation by an admin?